### PR TITLE
merge usability enhancement into `noetic`

### LINKED ISSUE #18: Highlight currently selected filter in filter list
The current implementation separates the `Filter Properties` pane and the `Filter List`. However, they should really be related, and the currently selected filter in the scene (highlighted in blue currently) should also be selected automatically in the filter list. This should improve visual clarity for selecting and editing filter settings/properties together.

### PR DESCRIPTION
**Pull request by PaperFanz**
This PR implements selection forwarding between filter list and filter scene for #18.

### CODE REVIEW

File: insitu/src/filter_graphics_item.cpp [hunk 1]
```diff
@@ -8,11 +9,13 @@ namespace insitu
...
 */
 FilterGraphicsItem::FilterGraphicsItem(boost::shared_ptr<insitu::Filter> filter,
+                                       QListWidgetItem* listItem,
                                        QGraphicsItem* parent)
     : QGraphicsItem(parent)
 {
-    this->filter = filter;
-    if (filter != nullptr)
+    filter_ = filter;
+    listItem_ = listItem;
```
Comment (AdamPettinger) [inline insitu/src/filter_graphics_item.cpp:17]: Could put these in the initializer list if you wanted

File: insitu/src/main.cpp [hunk 2]
```diff
@@ -13,11 +13,23 @@ int main(int argc, char** argv)
...
     QApplication app(argc, argv);
 
+    std::thread kill_thread([&app] {
+        ros::Rate loop_rate(4);
+        while (ros::ok())
+        {
+            loop_rate.sleep();
+        }
+        app.exit();
+    });
```
Comment (AdamPettinger) [inline insitu/src/main.cpp:23]: For what it is worth: You said this was fast and hacky but I have seen this approach before and used it in a GUI a while back

**Review by AdamPettinger — APPROVED** [review]
Looks good, nice work with Ctrl+C exiting and command line file path arg as well

File: insitu/src/main.cpp (hunk 2 above)
Comment (PaperFanz) [reply insitu/src/main.cpp:23]: I wasn't referring to this when I said fast and hacky; I think I lifted it directly from your code actually. The argument parsing on the other hand... has no error checking for now...

File: insitu/src/filter_graphics_item.cpp (hunk 1 above)
Comment (PaperFanz) [reply insitu/src/filter_graphics_item.cpp:17]: May leave this for a bigger "code cleanup" commit later down the line

**Pull request merged**